File: old_csharp_src/ShadowEmpire/SSEditOptionsWindowClass.cs
// Decompiled with JetBrains decompiler
// Type: WindowsApplication1.SSEditOptionsWindowClass
// Assembly: WindowsApplication1, Version=1.0.8020.28903, Culture=neutral, PublicKeyToken=null
// MVID: F52869E5-0850-48AD-BBBE-68E7A4900AFE
// Assembly location: C:\Program Files (x86)\Steam\steamapps\common\Shadow Empire\ShadowEmpire.exe

using System;
using System.Drawing;

namespace WindowsApplication1
{
  public class SSEditOptionsWindowClass : WindowClass
  {
    public int BBackId;
    public int BBackTextId;
    public int LibId;
    public int LibIdb;
    public int DashId;
    public int DashIdb;
    public int BackId;
    public int BackIdb;
    public int MapId;
    public int MapIdb;
    public int UnitId;
    public int UnitIdb;
    public int RegId;
    public int RegIdb;
    public int StringId;
    public int StringIdb;
    public int DebugId;
    public int DebugIdb;

    public void PopUpRefresh()
    {
    }

    public SSEditOptionsWindowClass(ref GameClass tGame, Bitmap screenbitmap = null, int sx = -1, int sy = -1)
      : base(ref tGame, tGame.ScreenWidth, 50, 9, screenbitmap: screenbitmap, sx: sx, sy: sy)
    {
      this.game.EditObj.inSimpleEditor = true;
      this.domenu();
    }

    public void domenu()
    {
      if (this.BackId > 0)
        this.RemoveSubPart(this.BackId);
      if (this.LibId > 0)
        this.RemoveSubPart(this.LibId);
      if (this.BackIdb > 0)
        this.RemoveSubPart(this.BackIdb);
      if (this.LibIdb > 0)
        this.RemoveSubPart(this.LibIdb);
      if (this.DashId > 0)
        this.RemoveSubPart(this.DashId);
      if (this.DashIdb > 0)
        this.RemoveSubPart(this.DashIdb);
      if (this.MapId > 0)
        this.RemoveSubPart(this.MapId);
      if (this.MapIdb > 0)
        this.RemoveSubPart(this.MapIdb);
      if (this.UnitId > 0)
        this.RemoveSubPart(this.UnitId);
      if (this.UnitIdb > 0)
        this.RemoveSubPart(this.UnitIdb);
      if (this.RegId > 0)
        this.RemoveSubPart(this.RegId);
      if (this.RegIdb > 0)
        this.RemoveSubPart(this.RegIdb);
      if (this.StringId > 0)
        this.RemoveSubPart(this.StringId);
      if (this.StringIdb > 0)
        this.RemoveSubPart(this.StringIdb);
      if (this.DebugId > 0)
        this.RemoveSubPart(this.DebugId);
      if (this.DebugIdb > 0)
        this.RemoveSubPart(this.DebugIdb);
      int num1 = 10 + (int) Math.Round((double) (this.game.ScreenWidth - 1024) / 2.0);
      SubPartClass tsubpart1 = (SubPartClass) new TextButtonPartClass("Exit", 100, tBackbitmap: (ref this.OwnBitmap), bbx: 10, bby: 5, theight: 40, usefont: this.game.MarcFont3, useshadow: true, tMarcStyle: true);
      this.BackId = this.AddSubPart(ref tsubpart1, 10, 5, 100, 40, 1);
      int num2 = num1 + 110;
      SubPartClass tsubpart2;
      if (this.game.EditObj.SimpleEditWindow != 95)
      {
        tsubpart2 = (SubPartClass) new TextButtonPartClass("Dashboard", 100, tBackbitmap: (ref this.OwnBitmap), bbx: num2, bby: 5, theight: 40, usefont: this.game.MarcFont3, useshadow: true, tMarcStyle: true);
        this.DashId = this.AddSubPart(ref tsubpart2, num2, 5, 100, 40, 1);
      }
      else
      {
        tsubpart2 = (SubPartClass) new TextButtonPartClass("Dashboard", 100, tBackbitmap: (ref this.OwnBitmap), bbx: num2, bby: 5, tinactive: true, theight: 40, usefont: this.game.MarcFont3, useshadow: true, tMarcStyle: true);
        this.DashIdb = this.AddSubPart(ref tsubpart2, num2, 5, 100, 40, 1);
      }
      int num3 = num2 + 110;
      if (this.game.EditObj.SimpleEditWindow != 101)
      {
        tsubpart2 = (SubPartClass) new TextButtonPartClass("Tables", 100, tBackbitmap: (ref this.OwnBitmap), bbx: num3, bby: 5, theight: 40, usefont: this.game.MarcFont3, useshadow: true, tMarcStyle: true);
        this.StringId = this.AddSubPart(ref tsubpart2, num3, 5, 100, 40, 1);
      }
      else
      {
        tsubpart2 = (SubPartClass) new TextButtonPartClass("Tables", 100, tBackbitmap: (ref this.OwnBitmap), bbx: num3, bby: 5, tinactive: true, theight: 40, usefont: this.game.MarcFont3, useshadow: true, tMarcStyle: true);
        this.StringIdb = this.AddSubPart(ref tsubpart2, num3, 5, 100, 40, 1);
      }
      int num4 = num3 + 110;
      if (this.game.EditObj.SimpleEditWindow != 122)
      {
        tsubpart2 = (SubPartClass) new TextButtonPartClass("Event Pics", 100, tBackbitmap: (ref this.OwnBitmap), bbx: num4, bby: 5, theight: 40, usefont: this.game.MarcFont3, useshadow: true, tMarcStyle: true);
        this.LibId = this.AddSubPart(ref tsubpart2, num4, 5, 100, 40, 1);
      }
      else
      {
        tsubpart2 = (SubPartClass) new TextButtonPartClass("Event Pics", 100, tBackbitmap: (ref this.OwnBitmap), bbx: num4, bby: 5, tinactive: true, theight: 40, usefont: this.game.MarcFont3, useshadow: true, tMarcStyle: true);
        this.LibIdb = this.AddSubPart(ref tsubpart2, num4, 5, 100, 40, 1);
      }
      int num5 = num4 + 110;
      if (this.game.EditObj.SimpleEditWindow != 123)
      {
        tsubpart2 = (SubPartClass) new TextButtonPartClass("Small Gfx", 100, tBackbitmap: (ref this.OwnBitmap), bbx: num5, bby: 5, theight: 40, usefont: this.game.MarcFont3, useshadow: true, tMarcStyle: true);
        this.UnitId = this.AddSubPart(ref tsubpart2, num5, 5, 100, 40, 1);
      }
      else
      {
        tsubpart2 = (SubPartClass) new TextButtonPartClass("Small Gfx", 100, tBackbitmap: (ref this.OwnBitmap), bbx: num5, bby: 5, tinactive: true, theight: 40, usefont: this.game.MarcFont3, useshadow: true, tMarcStyle: true);
        this.UnitIdb = this.AddSubPart(ref tsubpart2, num5, 5, 100, 40, 1);
      }
      int num6 = num5 + 110;
    }

    public override void DoRefresh() => this.domenu();

    public override WindowReturnClass HandleMouseClick(int x, int y, int b)
    {
      WindowReturnClass windowReturnClass = new WindowReturnClass();
      if (this.SubPartCounter > -1)
      {
        int subPartCounter = this.SubPartCounter;
        for (int index = 0; index <= subPartCounter; ++index)
        {
          if (x > this.SubPartX[index] & x < this.SubPartX[index] + this.SubPartW[index] && y > this.SubPartY[index] & y < this.SubPartY[index] + this.SubPartH[index])
          {
            int num = this.SubPartID[index];
            if (num == this.BackId)
            {
              this.game.EditObj.InEditor = false;
              if (!this.game.SuperAdminRights)
                this.game.EditObj.ShowInitialMenu = true;
              if (this.game.ModIntroType == 0)
                windowReturnClass.AddCommand(3, 1);
              else
                windowReturnClass.AddCommand(3, 12);
            }
            else
            {
              if (num == this.DashId)
              {
                this.game.EditObj.PencilMode = 0;
                windowReturnClass.AddCommand(1, 4);
                windowReturnClass.AddCommand(1, 5);
                windowReturnClass.AddCommand(2, 95);
                this.game.EditObj.SimpleEditWindow = 95;
                this.domenu();
                windowReturnClass.SetFlag(true);
                return windowReturnClass;
              }
              if (num == this.LibId)
              {
                this.game.EditObj.PencilMode = 0;
                windowReturnClass.AddCommand(1, 4);
                windowReturnClass.AddCommand(1, 5);
                windowReturnClass.AddCommand(2, 122);
                this.game.EditObj.SimpleEditWindow = 122;
                this.domenu();
                windowReturnClass.SetFlag(true);
                return windowReturnClass;
              }
              if (num == this.UnitId)
              {
                this.game.EditObj.PencilMode = 0;
                windowReturnClass.AddCommand(1, 4);
                windowReturnClass.AddCommand(1, 5);
                windowReturnClass.AddCommand(2, 123);
                this.game.EditObj.SimpleEditWindow = 123;
                this.domenu();
                windowReturnClass.SetFlag(true);
                return windowReturnClass;
              }
              if (num == this.DebugId)
              {
                this.game.EditObj.PencilMode = 0;
                windowReturnClass.AddCommand(1, 4);
                windowReturnClass.AddCommand(1, 5);
                windowReturnClass.AddCommand(2, 109);
                this.game.EditObj.SimpleEditWindow = 109;
                this.domenu();
                windowReturnClass.SetFlag(true);
                return windowReturnClass;
              }
              if (num == this.StringId)
              {
                this.game.EditObj.PencilMode = 0;
                windowReturnClass.AddCommand(1, 4);
                windowReturnClass.AddCommand(1, 5);
                windowReturnClass.AddCommand(2, 101);
                this.game.EditObj.SimpleEditWindow = 101;
                this.domenu();
                windowReturnClass.SetFlag(true);
                return windowReturnClass;
              }
            }
            return windowReturnClass;
          }
        }
        windowReturnClass.SetFlag(false);
        return windowReturnClass;
      }
      windowReturnClass.SetFlag(false);
      return windowReturnClass;
    }
  }
}

File: src/SSEditOptionsWindowClass.rs
// Decompiled with JetBrains decompiler
// Type: WindowsApplication1.SSEditOptionsWindowClass
// Assembly: WindowsApplication1, Version=1.0.8020.28903, Culture=neutral, PublicKeyToken=null
// MVID: F52869E5-0850-48AD-BBBE-68E7A4900AFE
// Assembly location: C:\Program Files (x86)\Steam\steamapps\common\Shadow Empire\ShadowEmpire.exe

// usingSystem;
// usingSystem.Drawing;

namespace WindowsApplication1
{
  pub class SSEditOptionsWindowClass : WindowClass
  {
    pub BBackId: i32;
    pub BBackTextId: i32;
    pub LibId: i32;
    pub LibIdb: i32;
    pub DashId: i32;
    pub DashIdb: i32;
    pub BackId: i32;
    pub BackIdb: i32;
    pub MapId: i32;
    pub MapIdb: i32;
    pub UnitId: i32;
    pub UnitIdb: i32;
    pub RegId: i32;
    pub RegIdb: i32;
    pub StringId: i32;
    pub StringIdb: i32;
    pub DebugId: i32;
    pub DebugIdb: i32;

    pub fn PopUpRefresh()
    {
    }

    pub SSEditOptionsWindowClass( tGame: GameClass, screenbitmap: Bitmap = null, let mut sx: i32 = -1, let mut sy: i32 = -1)
      : base( tGame, tGame.ScreenWidth, 50, 9, screenbitmap: screenbitmap, sx: sx, sy: sy)
    {
      self.game.EditObj.inSimpleEditor = true;
      self.domenu();
    }

    pub fn domenu()
    {
      if (self.BackId > 0)
        self.RemoveSubPart(self.BackId);
      if (self.LibId > 0)
        self.RemoveSubPart(self.LibId);
      if (self.BackIdb > 0)
        self.RemoveSubPart(self.BackIdb);
      if (self.LibIdb > 0)
        self.RemoveSubPart(self.LibIdb);
      if (self.DashId > 0)
        self.RemoveSubPart(self.DashId);
      if (self.DashIdb > 0)
        self.RemoveSubPart(self.DashIdb);
      if (self.MapId > 0)
        self.RemoveSubPart(self.MapId);
      if (self.MapIdb > 0)
        self.RemoveSubPart(self.MapIdb);
      if (self.UnitId > 0)
        self.RemoveSubPart(self.UnitId);
      if (self.UnitIdb > 0)
        self.RemoveSubPart(self.UnitIdb);
      if (self.RegId > 0)
        self.RemoveSubPart(self.RegId);
      if (self.RegIdb > 0)
        self.RemoveSubPart(self.RegIdb);
      if (self.StringId > 0)
        self.RemoveSubPart(self.StringId);
      if (self.StringIdb > 0)
        self.RemoveSubPart(self.StringIdb);
      if (self.DebugId > 0)
        self.RemoveSubPart(self.DebugId);
      if (self.DebugIdb > 0)
        self.RemoveSubPart(self.DebugIdb);
      let mut num1: i32 = 10 +  Math.Round( (self.game.ScreenWidth - 1024) / 2.0);
      let mut tsubpart1: SubPartClass =  new TextButtonPartClass("Exit", 100, tBackbitmap: ( self.OwnBitmap), bbx: 10, bby: 5, theight: 40, usefont: self.game.MarcFont3, useshadow: true, tMarcStyle: true);
      self.BackId = self.AddSubPart( tsubpart1, 10, 5, 100, 40, 1);
      let mut num2: i32 = num1 + 110;
      SubPartClass tsubpart2;
      if (self.game.EditObj.SimpleEditWindow != 95)
      {
        tsubpart2 =  new TextButtonPartClass("Dashboard", 100, tBackbitmap: ( self.OwnBitmap), bbx: num2, bby: 5, theight: 40, usefont: self.game.MarcFont3, useshadow: true, tMarcStyle: true);
        self.DashId = self.AddSubPart( tsubpart2, num2, 5, 100, 40, 1);
      }
      else
      {
        tsubpart2 =  new TextButtonPartClass("Dashboard", 100, tBackbitmap: ( self.OwnBitmap), bbx: num2, bby: 5, tinactive: true, theight: 40, usefont: self.game.MarcFont3, useshadow: true, tMarcStyle: true);
        self.DashIdb = self.AddSubPart( tsubpart2, num2, 5, 100, 40, 1);
      }
      let mut num3: i32 = num2 + 110;
      if (self.game.EditObj.SimpleEditWindow != 101)
      {
        tsubpart2 =  new TextButtonPartClass("Tables", 100, tBackbitmap: ( self.OwnBitmap), bbx: num3, bby: 5, theight: 40, usefont: self.game.MarcFont3, useshadow: true, tMarcStyle: true);
        self.StringId = self.AddSubPart( tsubpart2, num3, 5, 100, 40, 1);
      }
      else
      {
        tsubpart2 =  new TextButtonPartClass("Tables", 100, tBackbitmap: ( self.OwnBitmap), bbx: num3, bby: 5, tinactive: true, theight: 40, usefont: self.game.MarcFont3, useshadow: true, tMarcStyle: true);
        self.StringIdb = self.AddSubPart( tsubpart2, num3, 5, 100, 40, 1);
      }
      let mut num4: i32 = num3 + 110;
      if (self.game.EditObj.SimpleEditWindow != 122)
      {
        tsubpart2 =  new TextButtonPartClass("Event Pics", 100, tBackbitmap: ( self.OwnBitmap), bbx: num4, bby: 5, theight: 40, usefont: self.game.MarcFont3, useshadow: true, tMarcStyle: true);
        self.LibId = self.AddSubPart( tsubpart2, num4, 5, 100, 40, 1);
      }
      else
      {
        tsubpart2 =  new TextButtonPartClass("Event Pics", 100, tBackbitmap: ( self.OwnBitmap), bbx: num4, bby: 5, tinactive: true, theight: 40, usefont: self.game.MarcFont3, useshadow: true, tMarcStyle: true);
        self.LibIdb = self.AddSubPart( tsubpart2, num4, 5, 100, 40, 1);
      }
      let mut num5: i32 = num4 + 110;
      if (self.game.EditObj.SimpleEditWindow != 123)
      {
        tsubpart2 =  new TextButtonPartClass("Small Gfx", 100, tBackbitmap: ( self.OwnBitmap), bbx: num5, bby: 5, theight: 40, usefont: self.game.MarcFont3, useshadow: true, tMarcStyle: true);
        self.UnitId = self.AddSubPart( tsubpart2, num5, 5, 100, 40, 1);
      }
      else
      {
        tsubpart2 =  new TextButtonPartClass("Small Gfx", 100, tBackbitmap: ( self.OwnBitmap), bbx: num5, bby: 5, tinactive: true, theight: 40, usefont: self.game.MarcFont3, useshadow: true, tMarcStyle: true);
        self.UnitIdb = self.AddSubPart( tsubpart2, num5, 5, 100, 40, 1);
      }
      let mut num6: i32 = num5 + 110;
    }

    pub fn DoRefresh() => self.domenu();

    pub HandleMouseClick: WindowReturnClass(x: i32, y: i32, b: i32)
    {
      windowReturnClass: WindowReturnClass = WindowReturnClass::new();
      if (self.SubPartCounter > -1)
      {
        let mut subPartCounter: i32 = self.SubPartCounter;
        for (let mut index: i32 = 0; index <= subPartCounter; index += 1)
        {
          if (x > self.SubPartX[index] & x < self.SubPartX[index] + self.SubPartW[index] && y > self.SubPartY[index] & y < self.SubPartY[index] + self.SubPartH[index])
          {
            let mut num: i32 = self.SubPartID[index];
            if (num == self.BackId)
            {
              self.game.EditObj.InEditor = false;
              if (!self.game.SuperAdminRights)
                self.game.EditObj.ShowInitialMenu = true;
              if (self.game.ModIntroType == 0)
                windowReturnClass.AddCommand(3, 1);
              else
                windowReturnClass.AddCommand(3, 12);
            }
            else
            {
              if (num == self.DashId)
              {
                self.game.EditObj.PencilMode = 0;
                windowReturnClass.AddCommand(1, 4);
                windowReturnClass.AddCommand(1, 5);
                windowReturnClass.AddCommand(2, 95);
                self.game.EditObj.SimpleEditWindow = 95;
                self.domenu();
                windowReturnClass.SetFlag(true);
                return windowReturnClass;
              }
              if (num == self.LibId)
              {
                self.game.EditObj.PencilMode = 0;
                windowReturnClass.AddCommand(1, 4);
                windowReturnClass.AddCommand(1, 5);
                windowReturnClass.AddCommand(2, 122);
                self.game.EditObj.SimpleEditWindow = 122;
                self.domenu();
                windowReturnClass.SetFlag(true);
                return windowReturnClass;
              }
              if (num == self.UnitId)
              {
                self.game.EditObj.PencilMode = 0;
                windowReturnClass.AddCommand(1, 4);
                windowReturnClass.AddCommand(1, 5);
                windowReturnClass.AddCommand(2, 123);
                self.game.EditObj.SimpleEditWindow = 123;
                self.domenu();
                windowReturnClass.SetFlag(true);
                return windowReturnClass;
              }
              if (num == self.DebugId)
              {
                self.game.EditObj.PencilMode = 0;
                windowReturnClass.AddCommand(1, 4);
                windowReturnClass.AddCommand(1, 5);
                windowReturnClass.AddCommand(2, 109);
                self.game.EditObj.SimpleEditWindow = 109;
                self.domenu();
                windowReturnClass.SetFlag(true);
                return windowReturnClass;
              }
              if (num == self.StringId)
              {
                self.game.EditObj.PencilMode = 0;
                windowReturnClass.AddCommand(1, 4);
                windowReturnClass.AddCommand(1, 5);
                windowReturnClass.AddCommand(2, 101);
                self.game.EditObj.SimpleEditWindow = 101;
                self.domenu();
                windowReturnClass.SetFlag(true);
                return windowReturnClass;
              }
            }
            return windowReturnClass;
          }
        }
        windowReturnClass.SetFlag(false);
        return windowReturnClass;
      }
      windowReturnClass.SetFlag(false);
      return windowReturnClass;
    }
  }
}
